File: 021. Palindrome Check/PalindromeCheckOnO1.java
import java.util.*;

class PalindromeCheckOnO1 {
    public static boolean isPalindrome(String str) {
        int leftIndex = 0;
        int rightIndex = str.length() - 1;
        while (leftIndex < rightIndex) {
            if (str.charAt(leftIndex) != str.charAt(rightIndex)) {
                return false;
            }
            leftIndex++;
            rightIndex--;
        }
        return true;
    }
}

File: 021. Palindrome Check/PalindromeCheckOnO1.kt
package com.algoexpert.program

fun PalindromeCheckOnO1(string: String): Boolean {
    var leftIndex = 0
    var rightIndex = string.length - 1
    while (leftIndex < rightIndex) {
        if (string.get(leftIndex) != string.get(rightIndex)) {
            return false
        }
        leftIndex++
        rightIndex--
    }
    return true
}
